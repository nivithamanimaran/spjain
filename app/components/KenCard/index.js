import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';


const useStyles = makeStyles((theme) => ({
    root: {
        boxShadow: `0px 0px 9px ${theme.palette.KenColors.shadowColor}`,
        borderRadius: 3,
        padding: 40,
        marginTop: 10,
        width: '100%'
    },
    examTitle: {
        textAlign: "start",
        marginBottom: 20,
        fontSize: 14,
        color: "#193890fa",
        fontWeight: "bold"
    }
}));

export default function KenCard(props) {
    const { title } = props;

    const classes = useStyles();

    return (
        <React.Fragment>
            {/* <Typography className={classes.mainTitle}></Typography> */}
            <Paper elevation={0} className={classes.root} data-testid="KenCard">
                {title && <Typography className={classes.examTitle}>
                    {title}
                </Typography>}

                {props.children}
            </Paper>
        </React.Fragment>

    );
}
