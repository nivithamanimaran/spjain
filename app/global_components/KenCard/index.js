import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import { Box, useTheme } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  root: {
    boxShadow: `0px 0px 9px ${theme.palette.KenColors.shadowColor}`,
    borderRadius: 9,
    padding: 16,
    marginTop: 8,
    width: '100%',
    height: '100%',
  },
  examTitle: {
    textAlign: 'start',
    marginBottom: 20,
    fontSize: 14,
    color: '#193890fa',
    fontWeight: 'bold',
  },
  card: {
    padding: 16,
    marginTop: 10,
    width: '100%',
    border: `1px solid ${theme.palette.KenColors.neutral40}`,
    borderLeft: `3px solid ${theme.palette.KenColors.neutral60}`,
    borderRadius: 5,
  },
  cardInterTitle: {
    color: theme.palette.KenColors.primary,
    fontWeight: 600,
    fontSize: 14,
  },
}));

export default function KenCard(props) {
  const {
    title,
    paperStyles = {},
    activeBorderColor,
    activeBorderWidth = 2,
    active,
    variant,
  } = props;

  const classes = useStyles();
  const theme = useTheme();

  const getVariantStyles = () => {
    if (variant === 'plain') {
      return {};
    }
    return {};
  };

  const getCard = variant => {
    switch (variant) {
      case 'plain':
        return (
          <React.Fragment>
            <Paper
              className={`${classes.root} ${active ? classes.card : ''}`}
              style={{
                padding: 0,
                boxShadow: 'none',
                borderLeft: active
                  ? `${activeBorderWidth}px solid ${activeBorderColor ||
                      theme.palette.KenColors.primary}`
                  : '',
                ...paperStyles,
              }}
            >
              {title && (
                <Typography className={classes.examTitle}>{title}</Typography>
              )}

              {props.children}
            </Paper>
          </React.Fragment>
        );

      case 'cardInter':
        return (
          <React.Fragment>
            <div style={{ border: '1px solid #E3EBF6', borderRadius: 4 }}>
              {title && (
                <Box p={2} style={{ background: '#F1F5FA' }}>
                  <Typography className={classes.cardInterTitle}>
                    {title}
                  </Typography>
                </Box>
              )}

              <Box p={2}>{props.children}</Box>
            </div>
          </React.Fragment>
        );

      default:
        return (
          <React.Fragment>
            <Paper
              className={`${classes.root} ${active ? classes.card : ''}`}
              style={{
                borderLeft: active
                  ? `${activeBorderWidth}px solid ${activeBorderColor ||
                      theme.palette.KenColors.primary}`
                  : '',
                ...paperStyles,
              }}
            >
              {title && (
                <Typography className={classes.examTitle}>{title}</Typography>
              )}

              {props.children}
            </Paper>
          </React.Fragment>
        );
    }
  };

  return <>{getCard(variant)}</>;
}
