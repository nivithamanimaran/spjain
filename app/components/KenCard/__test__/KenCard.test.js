import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import ReactDOM from "react-dom";
import TestRenderer from "react-test-renderer";
import KenCard from "../index";
import ProvideTheme from "../../../Utils/helper/test/helperFunctions";

afterEach(() => {
  cleanup();
});

//checks that button renders or not
test("KenCard correctly renders", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <ProvideTheme>
      <KenCard />
    </ProvideTheme>,
    div
  );
});

//snapshot testing
it("matches snapshots", () => {
  const tree = TestRenderer.create( <ProvideTheme>
    <KenCard title="Title" /></ProvideTheme>).toJSON();
  expect(tree).toMatchSnapshot();
});
//testing with label
