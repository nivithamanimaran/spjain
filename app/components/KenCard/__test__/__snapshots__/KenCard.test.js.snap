// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-elevation0 MuiPaper-rounded"
  data-testid="KenCard"
>
  <p
    className="MuiTypography-root makeStyles-examTitle-2 MuiTypography-body1"
  >
    Title
  </p>
</div>
`;
